integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/distrox/DistroXEncryptedVolumeTest.java

```diff
@@ -1,77 +1,261 @@
+                .given(FreeIpaTestDto.class)
+                .then(this::verifyFreeIpaVolumeEncryptionKey)
+                .given(SdxTestDto.class)
+                    .withCloudStorage()
+                    .withExternalDatabase(sdxDatabaseRequest)
+                .when(sdxTestClient.create())
+                .await(SdxClusterStatusResponse.RUNNING)
+                .awaitForHealthyInstances()
+                .then(this::verifySdxVolumeEncryptionKey)
                 .given(DistroXTestDto.class)
-                .withInstanceGroupsEntity(distroXInstanceGroupTestDtos)
-                .withExternalDatabase(externalDatabaseName)
+                    .withInstanceGroupsEntity(distroXInstanceGroupTestDtos)
+                    .withExternalDatabase(distroXDatabaseRequest)
                 .when(distroXTestClient.create())
                 .await(STACK_AVAILABLE)
-                .then((tc, testDto, client) -> {
-                    sshEnaDriverCheckActions.checkEnaDriverOnAws(testDto.getResponse(), client);
-                    return testDto;
-                })
-                .then(new AvailabilityZoneAssertion())
+                .awaitForHealthyInstances()
+                .then(this::verifyAwsEnaDriver)
+                .then(new AwsAvailabilityZoneAssertion())
                 .then(validateTemplateContainsExternalDatabaseHostname())
-                .then((context, distrox, client) -> {
-                    distrox.getResponse();
-                    return distrox;
-                })
                 .validate();
     }
 
+    private DistroXTestDto verifyAwsEnaDriver(TestContext testContext, DistroXTestDto testDto, CloudbreakClient cloudbreakClient) {
+        if (CloudPlatform.AWS.equals(testDto.getCloudPlatform())) {
+            sshEnaDriverCheckActions.checkEnaDriverOnAws(testDto.getResponse(), cloudbreakClient);
+        } else {
+            LOGGER.warn(format("ENA driver is only available at AWS. So validation on '%s' provider is not possible!", testDto.getCloudPlatform()));
+        }
+        return testDto;
+    }
+
+    private EnvironmentTestDto verifyEnvironmentResponseDiskEncryptionKey(TestContext testContext, EnvironmentTestDto testDto,
+            EnvironmentClient environmentClient) {
+        DetailedEnvironmentResponse environment = environmentClient.getDefaultClient().environmentV1Endpoint().getByName(testDto.getName());
+        if (CloudPlatform.AWS.name().equals(environment.getCloudPlatform())) {
+            String encryptionKey = environment.getAws().getAwsDiskEncryptionParameters().getEncryptionKeyArn();
+            environmentName = testDto.getRequest().getName();
+            verifyDiskEncryptionKey(testDto.getCloudPlatform(), encryptionKey);
+        } else if (CloudPlatform.AZURE.name().equals(environment.getCloudPlatform())) {
+            String encryptionKey = environment.getAzure().getResourceEncryptionParameters().getEncryptionKeyUrl();
```

Comment:
For Azure, We should be checking if diskEncryptionSetId (DES)`environment.getAzure().getResourceEncryptionParameters().getDiskEncryptionSetId()` is present or not.

Azure encrypts the disks using DES instead of encryptionKey.

Reply:
Unfortunately in the Environment Response, the DES Id is not available, for example:
```"azure": {
    "resourceGroup": null,
    "resourceEncryptionParameters": {
      "encryptionKeyUrl": "https://cloud-jenkins-secrets.vault.azure.net/keys/key-azure-cmk-e2e/c1a1df3097f441b4a10cfcb7c8a3bcd9",
      "encryptionKeyResourceGroupName": "cloud-daily",
      "diskEncryptionSetId": null
    }
  }
```
Here we validate only the `encryptionKeyUrl` and `encryptionKeyResourceGroupName`. This was the cause why we introduced the FreeIPA creation to the test case previously. In case of FreeIPA and Datalake Response where we can validate, for example:
```
"dataDisks": [
        {
          "lun": 0,
          "name": "azuretestc0e39b3b-m-0-0-d41d8cd98f00b2",
          "createOption": "Attach",
          "caching": "ReadOnly",
          "managedDisk": {
            "diskEncryptionSet": {
              "id": "/subscriptions/3ddda1c7-d1f5-4e7b-ac81-0523f483b3b3/resourceGroups/AZURE-TEST-2A7DE84E6C2144339-CDP_DES-0C5EA00A-444E-4ADD-9225-E60A8FB1A20C/providers/Microsoft.Compute/diskEncryptionSets/azure-test-2a7de84e6c2144339-DES-0c5ea00a-444e-4add-9225-e60a8fb1a20c"
            },
            "storageAccountType": "StandardSSD_LRS",
            "id": "/subscriptions/3ddda1c7-d1f5-4e7b-ac81-0523f483b3b3/resourceGroups/azure-test-c0e39b3b3015/providers/Microsoft.Compute/disks/azuretestc0e39b3b-m-0-0-d41d8cd98f00b2"
          },
          "diskSizeGB": 100,
          "toBeDetached": false
        }
      ]
```
So we have two level of DES validation. The first is the environment where we validate the parameter for the encryption URL and later the second for the actual disks and theirs DES Id.